metrics/sinks/poseidon/driver.go

```diff
@@ -0,0 +1,114 @@
+package poseidon
+
+import (
+	"github.com/ICGog/poseidongo/pkg/stats"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"k8s.io/heapster/metrics/core"
+	"log"
+	"net/url"
+	"sync"
+)
+
+type PoseidonSink struct {
+	stats.PoseidonStatsClient
+	sync.RWMutex
+}
+
+func (sink *PoseidonSink) Name() string {
+	return "Poseidon Sink"
+}
+
+func (sink *PoseidonSink) Stop() {
+	// Do nothing.
+}
+
+func (sink *PoseidonSink) ExportData(dataBatch *core.DataBatch) {
+	sink.Lock()
+	defer sink.Unlock()
+
+	for metricKey, metricSet := range dataBatch.MetricSets {
+		if isNodeMetric(metricKey, metricSet) {
+			nodeStats := stats.NodeStats{
+				Hostname:       metricSet.Labels[core.LabelHostname.Key],
+				CpuAllocatable: getMetricValue(metricSet, core.MetricNodeCpuAllocatable).FloatValue,
+				CpuCapacity:    getMetricValue(metricSet, core.MetricNodeCpuCapacity).FloatValue,
+				CpuReservation: getMetricValue(metricSet, core.MetricNodeCpuReservation).FloatValue,
+				CpuUtilization: getMetricValue(metricSet, core.MetricNodeCpuUtilization).FloatValue,
+				MemAllocatable: getMetricValue(metricSet, core.MetricNodeMemoryAllocatable).FloatValue,
+				MemCapacity:    getMetricValue(metricSet, core.MetricNodeMemoryCapacity).FloatValue,
+				MemReservation: getMetricValue(metricSet, core.MetricNodeMemoryReservation).FloatValue,
+				MemUtilization: getMetricValue(metricSet, core.MetricNodeMemoryUtilization).FloatValue,
+			}
+			client, err := sink.ReceiveNodeStats(context.Background())
```

Comment:
Here you get back a NodeStatsResponse and err. Please change the code to:
nodeStatsResp, err := ...

Following, you can check the response and log appropriately:
1) See for the types of responses you can receive: https://github.com/ICGog/poseidongo/blob/master/pkg/stats/poseidonstats.proto
2) See how we handle the response in the Firmament client:
https://github.com/ICGog/poseidongo/blob/master/pkg/firmament/firmament_client.go#L59